### ISSUE TITLE
No way to pass custom name to `fileInfoAsync` through `resolveAsync`

### ISSUE THREAD

**Issue by EvanBacon** [expo/expo-asset-utils#2]

**Comment by spcotter** [comment]
Since a custom file name can't be specified, and remote assets are named from the URL by default, assets with the same name are overwritten. I'm trying to render map tiles in expo-three, and since tiles in the same row have the same name, they are being overwritten.

![img_b008729ef496-1](https://user-images.githubusercontent.com/1066220/38266415-9f1ff9b2-3746-11e8-9852-49143693d1c8.jpeg)

Passing a custom filename through `resolveAsset` and `resolveAsync` to `fromUriAsync` seems to do the trick.
```

const resolveAsset = async fileReference => {
  let urls = [];
  console.log('resolvingAsset', fileReference);
  if (Array.isArray(fileReference)) {
    for (let file of fileReference) {
      let asset;
      if (typeof file === 'string') {
        asset = await AssetUtils.resolveAsync(file);
      } else {
        asset = await AssetUtils.resolveAsync(file.reference, file.name);
      }
      urls.push(asset);
    }
  } else {
    console.log('typeof fileReference', typeof fileReference);
    let asset;
    if (typeof fileReference === 'string'){
      asset = await AssetUtils.resolveAsync(fileReference);
    } else {
      asset = await AssetUtils.resolveAsync(fileReference.reference, fileReference.name);
    }
    urls.push(asset);
  }
  return urls;
};
```

```
const resolveAsync = async (fileReference: WildCard, fileName:String): Promise<?Expo.Asset> => {
  if (fileReference instanceof Asset) {
    /// Asset
    if (!fileReference.localUri) {
      await fileReference.downloadAsync();
    }
    return fileReference;
  } else if (typeof fileReference === 'string') {
    /// uri
    const asset = await fromUriAsync(fileReference, fileName);
    asset.downloaded = true;
    return asset;
  } else if (typeof fileReference === 'number') {
    /// static resource
    const asset = await Asset.fromModule(fileReference);
    const output = await resolveAsync(asset, fileName);
    return output;
  } else if (isReactImageFormat(fileReference)) {
    /// { uri: string }
    const asset = await fromUriAsync(fileReference.uri, fileName);
    if (asset) {
      return await resolveAsync(asset, fileName);
    }
  }
};
```

![img_4d514ac89b36-1](https://user-images.githubusercontent.com/1066220/38266638-2c49185a-3747-11e8-8f53-47a56bf75dad.jpeg)

**Comment by EvanBacon** [comment]
# Fixed: `0.0.4-alpha.3`

```js
const asset = await AssetUtils.resolveAsync( "....", { fileName: "image.png" } );
```
